### ISSUE TITLE
System tests are fragile with runtimes other than crun or runc

### ISSUE THREAD

**Issue by dgibson** [containers/podman#12784]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The podman system tests (`./hack/bats`) in a number of places check for very specific output which comes from the low-level runtime.  This means that they're likely to fail with other low-level runtimes, even if those are operating perfectly correctly.

**Steps to reproduce the issue:**

1.  Configure `containers.conf` top select a low-level runtime other than runc or crun

2. Run the podman system tests `./hack/bats`

**Describe the results you received:**

Amongst others, some of the tests in `030-run.bats` fail, because the "no such file or directory" and/or "permission denied" messages don't exactly match the format used by runc/crun.

In my specific case, I was using a simple wrapper runtime around crun.  However, this fooled the script's logic detecting runc versus crun meaning it checked for runc format results, when it was emitting crun format results, again resulting in failure.

**Describe the results you expected:**

Tests pass.

**Output of `podman version`:**

```
$ ./bin/podman version
Client:       Podman Engine
Version:      4.0.0-dev
API Version:  4.0.0-dev
Go Version:   go1.16.12
Git Commit:   92f77c2440e75481e339d128aefebdeb92c546bc
Built:        Mon Jan 10 16:10:47 2022
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
$ CONTAINER_CONF=~/src/youki/test-containers.conf ./bin/podman info --debug
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.30-2.fc35.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.30, commit: '
  cpus: 8
  distribution:
    distribution: fedora
    version: "35"
  eventLogger: journald
  hostname: yekko
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.15.12-200.fc35.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 10055979008
  memTotal: 33424224256
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.4-1.fc35.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4
      commit: 3daded072ef008ef0840e8eccb0b52a7efbd165d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc35.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 66h 54m 0.74s (Approximately 2.75 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/dwg/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/dwg/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  volumePath: /home/dwg/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.0-dev
  Built: 1641791447
  BuiltTime: Mon Jan 10 16:10:47 2022
  GitCommit: 92f77c2440e75481e339d128aefebdeb92c546bc
  GoVersion: go1.16.12
  OsArch: linux/amd64
  Version: 4.0.0-dev
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

N/A - build from source

**Comment by dgibson** [comment]
I hope to supply a pull request address this shortly.

**Comment by dgibson** [comment]
I have draft patches for this at https://github.com/dgibson/podman/tree/ociplex-test
Note that they need some more work before I'm ready to submit a PR.

**Comment by vrothberg** [comment]
Thanks for reaching out and for starting work on it, @dgibson. Much appreciated!

**Comment by dgibson** [comment]
Addressed in #12958

**Comment by rhatdan** [comment]
Since that is merged, I am closing this Issue.
